### ISSUE TITLE
Support custom version strings on images with operator.

### ISSUE THREAD

**Issue by anish-db** [pingcap/tidb-operator#5213]
## Feature Request

**Is your feature request related to a problem? Please describe:**
We adapt the official pingcap images thru our internal docker image pipeline to produce images with custom version tags that don't follow the tidb version string format of `v6.5.3` etc.  e.g. our version strings look like `tidb_2023-06-27_00.43.39Z_master_123abcde_123456789`

Operator makes various assumptions to gate functionality based on wether a component is new enough past some version or not. For this it tries to parse the `.spec.version` field from the tidb cluster definition and errors out in our case like:

```
E0729 00:22:23.727017       1 utils.go:464] Parse version tidb_2023-06-27_00.43.39Z_master_123abcde_123456789 failure, error: Invalid Semantic Version
W0729 00:22:23.927754       1 tidb_member_manager.go:1102] parse tidb verson 'tidb_2023-06-27_00.43.39Z_master_123abcde_123456789' failed, skip setting store labels for TiKV of TiDB cluster tidb/devdev. err: Invalid Semantic Version
```

This makes a lot of error logs & prevents us from using various functionality and potentially incorrect usage at various operator code paths:
https://github.com/pingcap/tidb-operator/blob/34c19d4b977df6afe5f3a60db17905f71a2fc1cf/pkg/manager/member/tidb_member_manager.go#L1144
https://github.com/pingcap/tidb-operator/blob/34c19d4b977df6afe5f3a60db17905f71a2fc1cf/pkg/manager/member/utils.go#L463
https://github.com/pingcap/tidb-operator/blob/34c19d4b977df6afe5f3a60db17905f71a2fc1cf/pkg/manager/member/ticdc_scaler.go#L236

**Describe the feature you'd like:**
We would like to be able to specify an override version string for informing operator what tidb versions we are using which is separate from the actual image tag used for container images.

**Describe alternatives you've considered:**
N/A

**Teachability, Documentation, Adoption, Migration Strategy:**
Explained above.

cc: @coderplay @gegao-db

**Comment by csuzhangxc** [comment]
adding a new field as the desired version to override the version string from the image tag should work. But in some cases, if we need to know the current version of a Pod or StatefulSet, then we may need to get this information from HTTP APIs or store this information in the annotation of Pod/Statefulset.

**Comment by anish-db** [comment]
Yes, I think using version over HTTP Api in some cases would be fine since they would still be accurate in our case.

**Comment by adamf-db** [comment]
+1 to this feature request.

**Comment by Benjamin70598** [comment]
Adding a concrete failure mode and repro for this, hit on a production evaluation where the registry mandates digest-pinned references and does not serve tag-only pulls.

**Repro**

TidbCluster CR with any component image set to a digest-pinned reference, e.g.:

```yaml
tiflash:
  image: registry.example.com/pingcap/tiflash:v8.5.8@sha256:0000000000000000000000000000000000000000000000000000000000000000
  replicas: 4
```

Operator v1.6.x, TiFlash v8.5.8.

**Observed**

TiFlash pods CrashLoopBackOff at startup with:

```
Unknown setting use_uncompressed_cache
```

**Root cause**

`getImageVersion()` returns everything after the last colon, so a digest-pinned reference yields the digest hex rather than the tag. Semver parsing fails, and the error is discarded, so every version gate in the operator silently takes the old-version branch. For TiFlash specifically, `GetTiFlashConfig()` (pkg/manager/member/tiflash_util.go) gates on `>= v5.4.0`; on parse failure it falls through to the legacy config generator, which injects `profiles.default.use_uncompressed_cache = 0` via `SetIfNil`. That setting was removed from TiFlash in the 8.0 line, and in 8.5.x `Settings::setProfile()` calls `set()` unguarded, which throws `UNKNOWN_SETTING`, so TiFlash cannot start. Because the injection is `SetIfNil`, there is no CR-side way to remove the key.

The same parse failure also skips store label configuration ("parse tidb version ... failed, skip setting store labels") and affects every other version-gated code path, so the blast radius is wider than TiFlash config, just less visible.

Note on #5233: that approach resolves a digest as newer-than-any-version, which flips all gates to the new-version branch globally. A narrower alternative would be guarding the TiFlash config version check so an unparseable version emits no version-gated legacy config, similar in shape to the fix in #7007 for the TidbMonitor version parse.

Digest-pinned references (`name:tag@sha256:...`) are standard OCI format and increasingly mandated by supply-chain security policy, so this affects any environment with image-provenance requirements, not one unusual registry.
